packages/router/src/private-context.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { createContext, useCallback, useContext } from 'react'
+
+import { useRouterState } from './router-context'
+
+export interface PrivateState {
+  isPrivate: boolean
+  allowRender: (role?: string | string[]) => boolean
+  unauthenticated: string
```

Comment:
Could we add a docType here to let the user know what this is?

Reply:
I tried. Let me know how to improve, if needed.

Reply:
Nice, thanks!